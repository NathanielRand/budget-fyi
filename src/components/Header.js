import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { startLogout } from "../actions/auth";

export const Header = ({ startLogout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link to="/dashboard">
          <h1 className="header__title">
            Budget<b className="header__title-fyi">FYI</b>
          </h1>
        </Link>
        <button className="header__button" onClick={startLogout}>
          LOGOUT
        </button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
