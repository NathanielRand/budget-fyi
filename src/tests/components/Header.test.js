import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import Header from "../../components/Header";
import { getExpectedBodyHash } from "twilio/lib/webhooks/webhooks";

test("Should render Header correctly", () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Header />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
