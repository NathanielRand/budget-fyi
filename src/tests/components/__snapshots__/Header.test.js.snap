// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<header>
  <h1>
    Budget
    <b
      className="header-title-fyi"
    >
      FYI
    </b>
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Add Expense
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/about"
  >
    About
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/contact"
  >
    Contact
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
    Help
  </NavLink>
</header>
`;
